lib/hedge_fund_interview/interview_memory.ex

```diff
@@ -15,6 +15,22 @@ defmodule HedgeFundInterview.InterviewMemory do
     :ok
   end
 
+  def store_messages(messages) when is_list(messages) do
+    timestamp = :os.system_time(:millisecond)
+    entries =
+      messages
+      |> Enum.with_index()
+      |> Enum.map(fn {msg, index} ->
```

Comment:
No need to use `Enum.map`. Docs [HERE](https://hexdocs.pm/elixir/Enum.html#with_index/2).